File: src/absractions/data/ISyncable_Repository.ts
import { ISyncMetaData } from "../metadata/ISync_metadata";
import { IBaseRepository } from "./base_repository";

export interface ISyncableRepository<T> extends IBaseRepository<T> {
  updateMany(entities: T[]): Promise<T[]>;
  addMany(entities: T[]): Promise<T[]>;
  removeMany(entities: T[]): Promise<T[]>;
  fetchMany(metadata: ISyncMetaData):Promise<T[]>;
}

File: fast_sync_dart_client/fast_sync_client/lib/src/absraction/data/isyncable_repository.dart
import 'package:fast_sync_client/src/absraction/data/base_repository.dart';
import 'package:fast_sync_client/src/absraction/metadata/isync_metadata.dart';

abstract class ISyncableRepository<T> extends IBaseRepository<T> {
  Future<List<T>> updateMany(List<T> entities, ISyncMetadata metadata);
  Future<List<T>> addMany(List<T> entities, ISyncMetadata metadata);
  Future<List<T>> removeMany(List<T> entities, ISyncMetadata metadata);
  Future<List<T>> undirtyList(List<T> entities);
  Future<List<T>> processSyncResultForType(
      List<T> entities, ISyncMetadata metadata);
}
